extensions/2.0/Khronos/KHR_texture_astc/README.md

```diff
@@ -0,0 +1,240 @@
+
+When this extension is used, it's allowed to use value `image/ktx2` for the `mimeType` property of images that are referenced by the `source` property of `KHR_texture_astc` texture extension object.
+
+At runtime, engines can directly use the ktx images unlike KHR_texture_basisu which requires transcoding to other compressed formats. This also means you get support for all ASTC block formats and quality presets.
+
+## glTF Schema Updates
+
+The `KHR_texture_astc` extension is added to the `textures` object and specifies a `source` property that points to the index of the `image` which defines a reference to the KTX v2 file with the ASTC compressed image.
+
+The following glTF will load `astc_image.ktx2` in clients that support this extension, and fall back to `image.png` otherwise.
+
+```json
+{
+    "asset": {
+        "version": "2.0"
+    },
+    "extensionsUsed": [
+        "KHR_texture_astc"
+    ],
+    "textures": [
+        {
+            "source": 0,
+            "extensions": {
+                "KHR_texture_astc": {
+                    "source": 1
+                }
+            }
+        }
+    ],
+    "images": [
+        {
+            "uri": "image.png"
+        },
+        {
+            "uri": "astc_image.ktx2"
+        }
+    ]
+}
+```
+
+When used in the glTF Binary (GLB) format the `image` that points to the KTX v2 resource uses the `mimeType` value of `image/ktx2`.
+
+```json
+{
+    "asset": {
+        "version": "2.0"
+    },
+    "extensionsUsed": [
+        "KHR_texture_astc"
+    ],
+    "textures": [
+        {
+            "source": 0,
+            "extensions": {
+                "KHR_texture_astc": {
+                    "source": 1
+                }
+            }
+        }
+    ],
+    "images": [
+        {
+            "mimeType": "image/png",
+            "bufferView": 1
+        },
+        {
+            "mimeType": "image/ktx2",
+            "bufferView": 2
+        }
+    ]
+}
+```
+
+### Using Without a Fallback
+
+To use KTX v2 image with ASTC compression without a fallback, define `KHR_texture_astc` in both `extensionsUsed` and `extensionsRequired`. The `texture` object will then have its `source` property omitted as shown below.
+
+```json
+{
+    "asset": {
+        "version": "2.0"
+    },
+    "extensionsUsed": [
+        "KHR_texture_astc"
+    ],
+    "extensionsRequired": [
+        "KHR_texture_astc"
+    ],
+    "textures": [
+        {
+            "extensions": {
+                "KHR_texture_astc": {
+                    "source": 0
+                }
+            }
+        }
+    ],
+    "images": [
+        {
+            "uri": "astc_image.ktx2"
+        }
+    ]
+}
+```
+
+### JSON Schema
+
+[texture.KHR_texture_astc.schema.json](schema/texture.KHR_texture_astc.schema.json)
+
+## KTX v2 files with ASTC compression 
```

Comment:
Please update this section to reflect https://github.com/KhronosGroup/glTF/pull/2369#discussion_r1848786803.

Reply:
> Please update this section to reflect [#2369 (comment)](https://github.com/KhronosGroup/glTF/pull/2369#discussion_r1848786803).

@lexaknyazev Apologies there is a typo in that comment. 

> Agree, I realised specifiying normal maps themselves are a problem at the moment in glTF. I will remove HDR references from the extension for now.

What I meant to say was 

> Agree, I realised specifiying ~normal~**environment** maps themselves are a problem at the moment in glTF. I will remove HDR references from the extension for now.

In https://github.com/KhronosGroup/glTF/commit/8781d2296877799387f293c4ebc3da6bd0b9a254 I have not removed mentions of HDR and 3D but mentioned that these are not supported instead.

What's left in that section is the normal map stuff, which I didn't remove yet because waiting for your answer on https://github.com/KhronosGroup/glTF/pull/2369#discussion_r1848253882

The 2-component Normal maps are an important ASTC feature. To get the most quality for your normal maps from high-compression you need this. There is some other normal map specific optimisations in the encoder as well that you don't want to lose by just considering it like any other RGB color texture.